### PR TITLE
feat: import temperature and implement the simulation engine

### PR DESCRIPTION
**Pull request by ReidyT**
See #32 for a demo.

In this PR, I implemented a new iteration of the app by:
- importing the temperatures from CSV file (it could be hourly or daily measurement)
- implementing the mechanism to iterate over the temperatures (part of the simulation engine)
- implementing the heat loss and electricity cost
- changing the season according to the current period (range of dates)
- implementing unit tests to test the logic using vitest (no cypress tests for now)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,30 @@ If an existing model is updated, we should replace the type and the React compon
...
+## Downloading a CSV File
+
+To download a CSV file of temperatures, follow these steps:
+
+1. **CSV File Requirements**:
+
+   - The CSV file must contain a header.
+   - The first column should represent time in GMT+0, formatted in ISO 8601.
+   - The second column should contain the temperature values in °C.
+
```
Comment (spaenleh) [inline README.md:96]: Add an example file content below to illustrate. Juat a few lines is enough.

File: README.md [hunk 2]
```diff
@@ -84,6 +84,30 @@ If an existing model is updated, we should replace the type and the React compon
...
+2. **File Location**:
+
+   - Ensure that your CSV file is placed in the `public` directory of your project.
+   - For example, if your file is named `temperatures.csv`, the path will be `'temperatures.csv'` when referencing it from the public directory (i.e., `public/temperatures.csv`).
+
+3. **Data Source**:
+
+   - You can download the required CSV data from [Open Meteo](https://open-meteo.com/). Be aware of the measurement frequency, which can be either per hour or per day, based on historical data (per hour) or futur predictions (one temperature per day).
```
Comment (spaenleh) [inline README.md:104]:
```suggestion
   - You can download the required CSV data from [Open Meteo](https://open-meteo.com/). 
   
   > [!WARNING]
   > The measurement frequency changes depending on whether you are queriying historical data (one temperature per hour) or futur predictions (one temperature per day)
```

File: README.md [hunk 3]
```diff
@@ -84,6 +84,30 @@ If an existing model is updated, we should replace the type and the React compon
...
+
+4. **Using the TemperatureContext**:
```
Comment (spaenleh) [inline README.md:106]:
```suggestion
4. **Using the `TemperatureContext`**:
```

File: README.md [hunk 4]
```diff
@@ -84,6 +84,30 @@ If an existing model is updated, we should replace the type and the React compon
...
+   - In TemperatureContext, you can load this CSV file by referencing the appropriate path. Make sure to handle the data correctly based on the specified format and structure. Also be sure to specify the measurement frequency, which can be either per hour or per day, based on your needs.
```
Comment (spaenleh) [inline README.md:107]:
```suggestion
   - In `TemperatureContext`, you can load this CSV file by referencing the appropriate path. Make sure to handle the data correctly based on the specified format and structure. Also be sure to specify the measurement frequency, which can be either per hour or per day, based on your needs.
```

**Review by spaenleh — APPROVED** [review]
Looks very good ! 
I made a quick review of the changes, I like that you put a lot of unit tests !

Great work !

File: README.md [hunk 5]
```diff
@@ -84,6 +84,47 @@ If an existing model is updated, we should replace the type and the React compon
...
+
+   ```
+   time,temperature
+   1999-01-01,4.5
+   1999-01-02,3.6
+   ```
+
+2. **File Location**:
+
+   - Ensure that your CSV file is placed in the `public` directory of your project.
```
Comment (pyphilia) [inline README.md:113]:
```suggestion
   - Ensure that your CSV file is located in the `public` directory of your project.
```

File: README.md [hunk 6]
```diff
@@ -84,6 +84,47 @@ If an existing model is updated, we should replace the type and the React compon
...
+   - For example, if your file is named `temperatures.csv`, the path will be `'temperatures.csv'` when referencing it from the public directory (i.e., `public/temperatures.csv`).
+
+3. **Data Source**:
+
+   - You can download the required CSV data from [Open Meteo](https://open-meteo.com/).
+
+   > [!WARNING]
+   > The measurement frequency changes depending on whether you are queriying historical data (one temperature per hour) or futur predictions (one temperature per day).
```
Comment (pyphilia) [inline README.md:121]:
```suggestion
   > The measurement frequency changes depending on whether you are querying historical data (one temperature value per hour) or future predictions (one temperature value per day).
```

File: src/langs/en.json [hunk 7]
```diff
@@ -1,6 +1,16 @@
...
+    "YEARS_one": "{{count}} year",
+    "YEARS_other": "{{count}} years",
+    "MONTHS_one": "{{count}} month",
+    "MONTHS_other": "{{count}} months",
+    "WEEKS_one": "{{count}} week",
+    "WEEKS_other": "{{count}} weeks",
+    "DAYS_one": "{{count}} day",
+    "DAYS_other": "{{count}} days",
+    "HOURS_one": "{{count}} hour",
+    "HOURS_other": "{{count}} hours",
```
Comment (pyphilia) [inline src/langs/en.json:13]: Do you think you can use https://date-fns.org/ ? Can be in another PR.

File: src/modules/scenes/FirstScene.tsx [hunk 8]
```diff
@@ -1,64 +1,125 @@
...
+          variant="contained"
+          onClick={startSimulation}
+          disabled={status === SimulationStatus.RUNNING}
+        >
+          {status === SimulationStatus.RUNNING ? 'Running...' : 'Start'}
+        </Button>
+        <Stack spacing={1}>
+          <Stack>
+            <Typography variant="h6">
+              Simulation duration:{' '}
```
Comment (pyphilia) [inline src/modules/scenes/FirstScene.tsx:52]: Translate?

File: src/modules/scenes/FirstScene.tsx [hunk 9]
```diff
@@ -1,64 +1,125 @@
...
+              {t(TRANSLATIONS[duration.unit], { count: duration.value })}
+            </Typography>
+            <Stack pl={2}>
+              <Typography>Progression: {progression.progress}%</Typography>
```
Comment (pyphilia) [inline src/modules/scenes/FirstScene.tsx:56]: Translate?

File: src/modules/scenes/FirstScene.tsx [hunk 10]
```diff
@@ -1,64 +1,125 @@
...
+              <Typography>Time left: {progression.timeLeft}s</Typography>
+            </Stack>
+          </Stack>
+          <Stack>
+            <Typography>
+              Current period:{' '}
```
Comment (pyphilia) [inline src/modules/scenes/FirstScene.tsx:62]: Translate?

File: src/modules/scenes/FirstScene.tsx [hunk 11]
```diff
@@ -1,64 +1,125 @@
...
+              <Typography>Season: {season}</Typography>
+              <Typography>Outdoor (mean): {outdoorTemperature} °C</Typography>
+              <Typography>Indoor: {indoorTemperature} °C</Typography>
+              <Typography>
+                Heat Loss: {heatLoss.value} {heatLoss.unit}
+              </Typography>
+            </Stack>
+          </Stack>
+          <Typography>
+            Total Heat Loss: {totalHeatLoss.value} {totalHeatLoss.unit}
```
Comment (pyphilia) [inline src/modules/scenes/FirstScene.tsx:79]: Translate?

File: src/modules/scenes/FirstScene.tsx [hunk 12]
```diff
@@ -1,64 +1,125 @@
...
+          </Typography>
+          <Typography>Total Electricity cost: {electricityCost}.-</Typography>
```
Comment (pyphilia) [inline src/modules/scenes/FirstScene.tsx:81]: Translate?

**Review by pyphilia — APPROVED** [review]
It looks really well managed 👍

File: src/modules/scenes/FirstScene.tsx (hunk 8 above)
Comment (ReidyT) [reply src/modules/scenes/FirstScene.tsx:52]: I will remove this because it was just for debugging.
